frontend/src/components/productCard/ProductCard.tsx

```diff
@@ -57,33 +60,30 @@ export const ProductCard = ({ item }: ProductCardProps) => {
             </div>
             <div className={styles.avatarsAndUnits}>
                 <div className={styles.avatars}>
-                    {item.bidHistory.slice(-3).map((item) => {
-                        return (
-                            <div className={styles.avatar}>
-                                <ProfilePicture
-                                    url={item.bid.profileID.profilePicture}
-                                    width={"24px"}
-                                />
-                            </div>
-                        );
-                    })}
+                    {checkIsBidHistory() &&
+                        item.bidHistory.slice(-3).map((item, index) => {
+                            return (
+                                <div className={styles.avatar} key={index}>
+                                    <ProfilePicture
+                                        url={item.bid.profileID.profilePicture}
+                                        width={"24px"}
+                                    />
+                                </div>
+                            );
+                        })}
                 </div>
                 <span className={styles.unitsInStock}>{item.amount} in stock</span>
             </div>
             <div className={styles.bidSection}>
-                {item.bidHistory === []
-                    ? null
-                    : item.bidHistory.slice(-1).map((item) => {
-                          return (
-                              <span className={styles.highestBid}>
-                                  Highest bid
-                                  <span className={styles.highestBidValue}>
-                                      {item.bid.offer} ETH
-                                  </span>
-                              </span>
-                          );
-                      })}
-
+                {checkIsBidHistory() &&
+                    item.bidHistory.slice(-1).map((item) => {
```

Comment:
suggestion: you can move this into a variable instead of creating an array with one element and mapping over it.
`const highestBid = item.bidHistory.slice(-1)[0] // will be undefined or of type.

Also, i might be wrong here, but aren't you shadowing a prop, which is also named `item` in this map function?